### DISCUSSION TITLE
Hdd death loss file backup.channel

### DISCUSSION THREAD

**Discussion by HotCrypt** [lightningnetwork/lnd#8077] [Q&A] [answered]
Hi, My hdd is dead, I overwritten my backup.channel file (on usb key🤦‍♂️😭) without doing it on purpose. is it possible to recover my LND node on my pi4 with the seed phrase?
I also have the tx of the channels open(3) and all the informations.

I specify that I have already found links to recover the funds but I want to keep my node and continue with my existing channels.

Thank you.
Sincerely,

**Comment by HotCrypt** [comment]
I am currently running a data recovery software to try to recover the channel.backup file. The software renames the files found. 
File .backup is ok for rescherch Or do I have to look for another type of file?

**Reply by guggero** [reply, marked as answer]
Yes, look for `channel.backup`. Then follow the instructions here, if you found it: https://github.com/lightningnetwork/lnd/blob/master/docs/recovery.md
If you can't restore that file, it gets tricky. Then a lot of manual work is required to close the channels. See https://github.com/lightninglabs/chantools#when-should-i-use-what-command for more information.

**Reply by HotCrypt** [reply]
Ok thank you Sir 👍

**Comment by bitromortac** [comment]
Seeing https://github.com/lightningnetwork/lnd/issues/8122 and https://github.com/lightninglabs/chantools/discussions/89, could you describe the steps you took so far? I guess you are trying to start some manual recovery process?

**Reply by HotCrypt** [reply]
Yes, successfully completed.

Everything works perfectly, it's just a mistake with a peer.

He is no longer often online I think:

`lncli disconnect 03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7`
[lncli] rpc error: code = Unknown desc = unable disconnect peer: peer 03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7 is not connected

**Reply by bitromortac** [reply]
That is normal behavior, if the peer is not connected, you can't disconnect. What's the reason you try to disconnect?

**Reply by HotCrypt** [reply]
I managed to disconnect from this node this morning (successful { }) but I still get the errors:

`//*2023-11-04 16:47:28.185 
[INF] SRVR: New inbound connection from 165.227.101.132:34990
[INF] SRVR: Finalizing connection to 03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7@165.227.101.132:34990, inbound=true

[INF] NTFN: New block epoch subscription
[INF] PEER: Peer(03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7): Negotiated chan series queries
[INF] DISC: Creating new GossipSyncer for peer=03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7
[INF] DISC: GossipSyncer(03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7): applying gossipFilter(start=2023-11-04 16:47:28.1870146 +0100 CET m=+36086.440853714, end=2159-12-11 23:15:43.1870146 +0100 CET)
[ERR] PEER: Peer(03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7): resend failed: unable to fetch channel sync messages for peer 03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7@165.227.101.132:34990: unable to find closed channel summary
[INF] SRVR: New inbound connection from 185.220.101.139:33829
[INF] SRVR: Finalizing connection to 0225a3f8e57be37f0e46e8af581eb8c4c2f1ecfb53b97a7c01488ef529c79972ac@185.220.101.139:33829, inbound=true
[INF] PEER: Peer(0225a3f8e57be37f0e46e8af581eb8c4c2f1ecfb53b97a7c01488ef529c79972ac): Negotiated chan series queries
[INF] DISC: Creating new GossipSyncer for peer=0225a3f8e57be37f0e46e8af581eb8c4c2f1ecfb53b97a7c01488ef529c79972ac
[INF] DISC: GossipSyncer(0225a3f8e57be37f0e46e8af581eb8c4c2f1ecfb53b97a7c01488ef529c79972ac): applying gossipFilter(start=2023-11-04 16:47:44.272399007 +0100 CET m=+36102.526238121, end=2159-12-11 23:15:59.272399007 +0100 CET)
[INF] NTFN: New block epoch subscription
[INF] WTCL: (legacy) Client stats: tasks(received=0 accepted=0 ineligible=0) sessions(acquired=0 exhausted=0)
[INF] WTCL: (anchor) Client stats: tasks(received=2 accepted=0 ineligible=2) se`


why is lnd trying to sync with it again? I just want to delete it and not get any more error messages.
I also specify that I have a channel open with this node but it is now lost because I recovered my node without channel.backup. Afterwards I just connected to it without opening a channel, and since then I have had these error messages.

Thx again for your help.

**Reply by bitromortac** [reply]
Does this channel still appear in pending channels, could you post that entry? It looks like the peer still thinks there is a channel and tries to connect to your node to reestablish it.

**Reply by HotCrypt** [reply]
> Does this channel still appear in pending channels, could you post that entry? It looks like the peer still thinks there is a channel and tries to connect to your node to reestablish it.

Yes of course:

**Reply by HotCrypt** [reply]
`~ $ lncli pendingchannels
{
   
"total_limbo_balance": "0",
    "pending_open_channels": [],
    "pending_closing_channels": [],
    "pending_force_closing_channels": [],
    "waiting_close_channels": []
}
 `

**Comment by HotCrypt** [comment]


**Reply by bitromortac** [reply]
So you kept reusing your old node after recovery (you didn't create a new wallet?). Do you have that peer in your channel list (`listchannels`)?

**Reply by HotCrypt** [reply]
-Yes, he's in listpeers even if I disconnected from him.

_Yes, I kept the same wallet, hoping that I can recover (700k.sat off-chain) without a backup file, otherwise it doesn't matter.

For my error peers: I think LND remembers that I had an open channel (500ksat) with this node and the fact that I connected to it without recovering the old channel makes it buggy it seems 🤔.

I disconnected from him and he is still present in my peers, strange.

Maybe I didn't have to connect to him until I closed the open channel with this node.

**Reply by bitromortac** [reply]
Ok, so there is no channel left in `listchannels` or `pendingchannels` for this peer? Can you find the channel in `closedchannels`? The peer should not connect to your node if all the channels have resolved, so that tells us that they consider it still open. We would need to figure out what status the channel has on your node. Can you find the channel point for that problematic 500ksat channel?

**Reply by HotCrypt** [reply]
- yes exactly for listchan and pendingchan, anyway I only have one channel open with ln.plus (20ksat) for the moment.
-yes no problem:
Channel Point 2077a89e5b9d3685ff84eb856fb9341b492f53ec04f135268808c74cf0d0ef88:1

Thx again bitromortac

**Reply by bitromortac** [reply]
Ok, that peer seems to still be online, it is strange that the static channel backup couldn't restore this channel (perhaps the peer changed its address in the meantime). So there is a chance that the funds can be recovered. For now I would recommend to start with a new lnd instance and wallet if you need to use the Lightning network. Do you have your old channel backup file when the channels were open?

**Comment by HotCrypt** [comment]
Solved. I no longer have an error with the peer.
Thank you all for your help.

**Reply by HotCrypt** [reply]
Please excuse me for reopening this topic but my node is still trying to connect with node 03df, I think I found why: it changed its ip address, here is its old address with which I had to open a channel (now chan lost): 

03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7@165.227.101.132:41178 (Port change,why?On the logs the port changes too)

Today his IP has changed.

Is there a way to ban or remove a peer so that my node no longer tries the gossip peer and thus avoid repetitive (and unnecessary) errors with this peer?

I managed to disconnect but it always comes back with the old ip:

> 2023-11-15 01:05:45.486 [INF] PEER: Peer(03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7): disconnecting 03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7@165.227.101.132:58562, reason: read handler closed
2023-11-15 01:05:45.486 [INF] NTFN: Cancelling epoch notification, epoch_id=765
2023-11-15 01:05:45.587 [INF] DISC: Removing GossipSyncer for peer=03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7
2023-11-15 01:05:58.749 [INF] WTCL: (legacy) Client stats: tasks(received=0 accepted=0 ineligible=0) sessions(acquired=0 exhausted=0)
2023-11-15 01:05:58.862 [INF] WTCL: (anchor) Client stats: tasks(received=145 accepted=119 ineligible=145) sessions(acquired=0 exhausted=0)
2023-11-15 01:06:41.309 [INF] CRTR: Processed channels=0 updates=89 nodes=17 in last 1m7.029572715s
2023-11-15 01:06:58.749 [INF] WTCL: (legacy) Client stats: tasks(received=0 accepted=0 ineligible=0) sessions(acquired=0 exhausted=0)
2023-11-15 01:06:58.861 [INF] WTCL: (anchor) Client stats: tasks(received=145 accepted=119 ineligible=145) sessions(acquired=0 exhausted=0)
2023-11-15 01:07:34.475 [INF] CRTR: Processed channels=0 updates=84 nodes=8 in last 53.166507185s
2023-11-15 01:07:58.749 [INF] WTCL: (legacy) Client stats: tasks(received=0 accepted=0 ineligible=0) sessions(acquired=0 exhausted=0)
2023-11-15 01:07:58.861 [INF] WTCL: (anchor) Client stats: tasks(received=145 accepted=119 ineligible=145) sessions(acquired=0 exhausted=0)
2023-11-15 01:08:34.279 [INF] CRTR: Processed channels=0 updates=161 nodes=18 in last 59.803685504s
2023-11-15 01:08:48.482 [INF] SRVR: New inbound connection from 165.227.101.132:39692
2023-11-15 01:08:48.482 [INF] SRVR: Finalizing connection to 03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7@165.227.101.132:39692, inbound=true
2023-11-15 01:08:48.483 [INF] PEER: Peer(03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7): Negotiated chan series queries
2023-11-15 01:08:48.483 [INF] DISC: Creating new GossipSyncer for peer=03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7
2023-11-15 01:08:48.483 [INF] NTFN: New block epoch subscription
2023-11-15 01:08:48.581 [ERR] PEER: Peer(03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7): resend failed: unable to fetch channel sync messages for peer 03df3f0a2fd6bea5429a596461ce784c922b2981ada1af89cfefcd9ccfb16c16a7@165.227.101.132:39692: unable to find closed channel summary

I really don't like having error messages in the logs.

Thx again. 
Cordialy,

**Reply by bitromortac** [reply]
So the peer still tries to reach you to reestablish the channel, so in principle you should be able to recover the funds. I'm not sure why the channel backup didn't work. It could be that the channel backup had an old ip address and couldn't reach the peer for the recovery process. Perhaps it's possible to modify the channel backup to use the new ip. The easiest way would be to get in contact with the other node and ask it to force close, otherwise a fake channel backup could help, but I have never tried that.

**Reply by HotCrypt** [reply]
Ok thanks, I will send him a message so that he forces the closure of the channel.

For recovery, I did not keep any backup files, I only have the seed phrase. 
I try a recovery with chantools later in the week.

**Comment by HotCrypt** [comment]
Thank you all for your help, I managed to close the channel thanks to Oliver tutorial (very good job) and I no longer have any errors now.

Sincerely,
